production/tanka/grafana-agent/config.libsonnet

```diff
@@ -70,9 +70,9 @@
 
           scrape_configs:
             if $._config.agent_host_filter then
-              $._config.kubernetes_scrape_configs + $._config.deployment_scrape_configs
+              $._config.kubernetes_scrape_configs
             else
-              $._config.kubernetes_scrape_configs,
+              $._config.kubernetes_scrape_configs + $._config.deployment_scrape_configs,
```

Comment:
This was the bug. When host filtering is enabled, it shouldn't include the deployment scrape configs.